registrar/registrar.go

```diff
@@ -87,23 +91,27 @@ func (r *registrar) Run(signals <-chan os.Signal, ready chan<- struct{}) error {
 	healthyChan := make(chan config.Route, len(r.config.Routes))
 	unhealthyChan := make(chan config.Route, len(r.config.Routes))
 
-	periodicHealthcheckCloseChans := make([]chan struct{}, len(r.config.Routes))
+	periodicHealthcheckCloseChans := &PeriodicHealthcheckCloseChans{}
 
-	for i := range periodicHealthcheckCloseChans {
-		periodicHealthcheckCloseChans[i] = make(chan struct{}, len(r.config.Routes))
-	}
+	for _, route := range r.config.Routes {
+		closeChan := periodicHealthcheckCloseChans.Add(route)
 
-	for i, route := range r.config.Routes {
 		go r.periodicallyDetermineHealth(
 			route,
 			nohealthcheckChan,
 			errChan,
 			healthyChan,
 			unhealthyChan,
-			periodicHealthcheckCloseChans[i],
+			closeChan,
 		)
 	}
 
+	routeDiscovered := make(chan config.Route)
+	routeRemoved := make(chan config.Route)
+
+	routesConfigWatcher := NewRoutesConfigWatcher(r.logger, r.dynamicConfigDiscoveryInterval, r.config.DynamicConfigGlobs, routeDiscovered, routeRemoved)
+	routesConfigWatcherProcess := ifrit.Background(routesConfigWatcher)
```

Comment:
Should we make running this conditional on whether or not there are any DynamicConfigGlobs to watch?